### DISCUSSION TITLE
Modders, when changing characters mid-song, remember to preload them at the start of the song before changing to them!

### DISCUSSION THREAD

**Discussion by biomseed** [FunkinCrew/Funkin#2347] [General]
This can be done by using Paths.getSparrowAtlas on the image of the character, or any other images the character might pull from, at the start of your song. Alternatively use Paths.animateAtlas if the image is an adobe animate texture atlas. These functions not only return the sprites, they make the game load the images into the game. If you don't do this, then the character that you change to will have to load whenever you change to it if it hasn't been loaded already before, causing lag. A lot of mods don't do this and some mods for this update are already not doing this, so I'm just letting you know if you're new to this stuff.
The first character in the song that you pick in the chart itself is already loaded at the start of the song, so you don't need to do anything for that.

**Comment by Johferson** [comment]
How do you do this but for videos?

**Reply by biomseed** [reply]
> How do you do this but for videos?

I'm pretty sure it's Paths.videos, but I'm not entirely sure if it will work. It most likely will though.

**Comment by mayo78** [comment]
its even better if you just create the character object before using it instead of creating a new object when you need it

**Reply by biomseed** [reply]
you don't have to create a new object to change characters
